tests/system/src/test/java/test/javafx/scene/web/StraightLineTest.java

```diff
@@ -0,0 +1,176 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene.web;
+
+import com.sun.webkit.WebPage;
+import com.sun.webkit.WebPageShim;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.image.PixelReader;
+import javafx.scene.image.WritableImage;
+import javafx.scene.paint.Color;
+import javafx.scene.web.WebEngineShim;
+import javafx.scene.web.WebView;
+import javafx.stage.Stage;
+import java.io.*;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import test.util.Util;
+
+import java.util.concurrent.CountDownLatch;
+import static org.junit.Assert.assertNotEquals;
+
+import static javafx.concurrent.Worker.State.SUCCEEDED;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class StraightLineTest {
+    private static final CountDownLatch launchLatch = new CountDownLatch(1);
+
+    // Maintain one application instance
+    static StraightLineTestApp straightLineTestApp;
+
+    private WebView webView;
+
+    public static class StraightLineTestApp extends Application {
+        Stage primaryStage = null;
+
+        @Override
+        public void init() {
+            StraightLineTest.straightLineTestApp = this;
+        }
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            Platform.setImplicitExit(false);
+            this.primaryStage = primaryStage;
+            this.primaryStage.setWidth(80);
+            this.primaryStage.setHeight(60);
+            launchLatch.countDown();
+        }
+    }
+    private static String colorToString(Color c) {
+        int r = (int)(c.getRed() * 255.0);
+        int g = (int)(c.getGreen() * 255.0);
+        int b = (int)(c.getBlue() * 255.0);
+        int a = (int)(c.getOpacity() * 255.0);
+        return "rgba(" + r + "," + g + "," + b + "," + a + ")";
+    }
+
+
+    @BeforeClass
+    public static void setupOnce() {
+        // Start the Test Application
+        new Thread(() -> Application.launch(StraightLineTestApp.class, (String[])null)).start();
+
+        assertTrue("Timeout waiting for FX runtime to start", Util.await(launchLatch));
+    }
+
+    @AfterClass
+    public static void tearDownOnce() {
+        Platform.exit();
+    }
+
+    @Before
+    public void setupTestObjects() {
+        Platform.runLater(() -> {
+            webView = new WebView();
+            Scene scene = new Scene(webView,80,60);
```

Comment:
space added, i will look this observation on windows

Reply:
I can confirm that even on Mac, the only reason that this test passes -- and it does so either with or without the fix -- is because the window size is too small. You need to make the Scene larger. I recommend this:

```
    new Scene(webView, 200, 150);
```

It's generally better to set the size of the scene rather than the stage for tests like this -- especially since you are taking a snapshot of the scene.